packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs

```diff
@@ -22,38 +22,62 @@ use crate::{
 pub trait PersistedOntologyType {
     type Inner;
 
-    fn new(inner: Self::Inner, identifier: PersistedOntologyIdentifier) -> Self;
+    fn from_record(record: OntologyRecord<Self::Inner>) -> Self;
 }
 
 impl PersistedOntologyType for PersistedDataType {
     type Inner = DataType;
 
-    fn new(inner: Self::Inner, identifier: PersistedOntologyIdentifier) -> Self {
-        Self { inner, identifier }
+    fn from_record(data_type: OntologyRecord<Self::Inner>) -> Self {
```

Comment:
This makes me wonder if we really want the `OntologyRecord` struct _and_ the `Persisted*` structs..

Probably fine for now because it gives us a good place to put the component macros and such.

Reply:
We could hide the `is_latest` field on `OntologyRecord` and use it everywhere we use `Persisted*`, but I think this is a good change for another PR 🙂 

I expect the API to be cleaned up by a lot when we remove the old endpoints, it's probably worth doing it at that step.